application/src/main/java/org/togetherjava/tjbot/commands/basic/RoleSelectCommand.java

```diff
@@ -0,0 +1,262 @@
+            .build();
+    }
+
+    @Override
+    public void onSelectionMenu(@NotNull SelectionMenuEvent event, @NotNull List<String> args) {
+        Member member = Objects.requireNonNull(event.getMember(), "Member is null");
+
+        Guild guild = Objects.requireNonNull(event.getGuild());
+
+        List<Role> selectedRoles = new ArrayList<>();
+        for (SelectOption selectedOption : Objects.requireNonNull(event.getSelectedOptions())) {
+            Role selectedRole = guild.getRoleById(selectedOption.getValue());
+            if (selectedRole != null && guild.getSelfMember().canInteract(selectedRole)) {
+                selectedRoles.add(selectedRole);
+            }
+        }
+
+        // True if the event option was 'choose'
+        if (event.getMessage().isEphemeral()) {
+
+            SelectionMenu.Builder menu = SelectionMenu.create(generateComponentId(member.getId()));
+            menu.setPlaceholder("Select your roles")
+                .setMaxValues(selectedRoles.size())
+                .setMinValues(0);
+
+            // Add selected options to the menu
+            selectedRoles.forEach(role -> menu.addOption(role.getName(), role.getId()));
+
+            event.getChannel()
+                .sendMessageEmbeds(event.getMessage().getEmbeds().get(0))
+                .setActionRow(menu.build())
+                .queue();
+
+            event.reply("Message sent successfully!").setEphemeral(true).queue();
+
+            return;
+        }
+
+        List<SelectOption> menuOptions =
+                Objects.requireNonNull(event.getInteraction().getComponent()).getOptions();
+
+
+        for (SelectOption selectedOption : menuOptions) {
+            Role role = guild.getRoleById(selectedOption.getValue());
+
+            if (role == null) {
+                logger.info(
+                        "The {} ({}) role has been removed but is still an option in the selection menu",
+                        selectedOption.getLabel(), selectedOption.getValue());
+                continue;
+            }
+
+            if (selectedRoles.contains(role)) {
+                guild.addRoleToMember(member, role).queue();
+            } else {
+                event.getGuild().removeRoleFromMember(member, role).queue();
+            }
+        }
+
+        event.reply("Updated your roles!").setEphemeral(true).queue();
+    }
+
+    /**
+     * This gets the OptionMapping and returns the value as a string if there is one
+     *
+     * <p>
+     * </p>
+     *
+     * @param option the {@link OptionMapping}
+     * @return the value. nullable {@link String}
+     */
+    private static @Nullable String handleOption(@Nullable OptionMapping option) {
+
+        if (option == null) {
+            return null;
+        }
+
+        if (option.getType() == OptionType.STRING) {
+            return option.getAsString();
+        } else if (option.getType() == OptionType.BOOLEAN) {
+            return option.getAsBoolean() ? "true" : "false";
```

Comment:
```suggestion
            return option.getAsBoolean().toString();
```

Not sure, but I'd assume toStringing would also result in `true` or `false`